lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,460 @@
+package azure
+
+import (
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/lacework/go-sdk/lwgenerate"
+	"github.com/pkg/errors"
+)
+
+type GenerateAzureTfConfigurationArgs struct {
+	// Should we configure Activity Log integration in LW?
+	ActivityLog bool
+
+	// Should we add Config integration in LW?
+	Config bool
+
+	// Should we create an Active Directory integration
+	AdIntegration bool
+
+	// If Config is true, give the user the opportunity to name their integration. Defaults to "TF Config"
+	ConfigIntegrationName string
+
+	// If ActivityLog is true, give the user the opportunity to name their integration. Defaults to "TF activity log"
+	ActivityLogIntegrationName string
+
+	// Active Directory application Id
+	AdApplicationId string
+
+	// Active Directory password
+	AdApplicationPassword string
+
+	// Active Directory Enterprise app object id
+	AdServicePrincipalId string
+
+	// Should we use the management group, rather than subscription
+	ManagementGroup bool
+
+	// Management Group ID to set
+	ManagementGroupId string
+
+	// List of subscription Ids
+	SubscriptionIds []string
+
+	// Grant read access to ALL subscriptions
+	AllSubscriptions bool
+
+	// Storage Account name
+	StorageAccountName string
+
+	// Storage Account Resource Group
+	StorageAccountResourceGroup string
+
+	// Should we use existing storage account
+	ExistingStorageAccount bool
+
+	// Azure region where the storage account for loggin resides
+	Location string
+}
+
+// Ensure all combinations of inputs are valid for supported spec
+func (args *GenerateAzureTfConfigurationArgs) validate() error {
+	// Validate one of config or activity log was enabled; otherwise error out
+	if !args.ActivityLog && !args.Config {
+		return errors.New("audit log or config integration must be enabled")
+	}
+
+	// Validate that active directory settings are correct
+	if !args.AdIntegration {
```

Comment:
Should this be a not check on `arg.AdIntegration`?

Additionally, it seems to me that this could be flattened to a single if
```
if args.AdIntegration && (args.AdApplicationId == "" || args.AdServicePrincipalId == "" || args.AdApplicationPassword == "")
```

Reply:
No in this case AdIntegration means create an AD integration and use the generated details from terraform, probably should change the name, in the case where this is not set then we need to enter the details of an already existing AD integration. 
Could make it a single if statement